app/controllers/ops_controller/ops_rbac.rb

```diff
@@ -1373,13 +1376,18 @@ def rbac_role_add_parent(node)
   # Set role record variables to new values
   def rbac_role_set_record_vars(role)
     role.name = @edit[:new][:name]
-    role.settings ||= {}
+    role.settings ||= {:restrictions => {}}
     if @edit[:new][:vm_restriction] == :none
-      role.settings.delete(:restrictions)
+      role.settings[:restrictions].delete(:vms)
     else
-      role.settings[:restrictions] = {:vms => @edit[:new][:vm_restriction]}
+      role.settings[:restrictions][:vms] = @edit[:new][:vm_restriction]
     end
-    role.settings = nil if role.settings.empty?
+    if @edit[:new][:service_template_restriction] == :none
+      role.settings[:restrictions].delete(:service_templates)
+    else
+      role.settings[:restrictions][:service_templates] = @edit[:new][:service_template_restriction]
+    end
+    role.settings = nil if role.settings == {:restrictions => {}}
```

Comment:
```suggestion
    role.settings = nil if role.settings[:restrictions].empty?
```

Reply:
is .blank? better in case restrictions is not set yet?